app/queries/get_grouped_content_and_links.rb

```diff
@@ -0,0 +1,116 @@
+module Queries
+  class GetGroupedContentAndLinks
```

Comment:
Would it be interesting to explain the reason for using the `content_id` in pagination here?